### PR TITLE
Implementation of CSR-DCF tracker

### PR DESCRIPTION
**Pull request by amuhic**
This pullrequest implements CSR-DCF tracker based on: 
http://prints.vicos.si/publications/355
http://prints.vicos.si/publications/files/355

I also have one question. In modules/tracking/src/trackerCSRTUtils.cpp, I've copied function:
``void computeHOG32D(const Mat &imageM, Mat &featM, const int sbin, const int pad_x, const int pad_y)
`` from modules/dpm/src/dpm_feature.cpp (with some minor changes). This was mostly used for testing purposes, to see whether this implementation of FHOG could be used in this tracker. I'm asking for your advice on how to go on about this. I'm assuming that a dependency for dpm module should be added to tracking module and then use that function directly.

```
force_builders=Mac
```

### CODE REVIEW

File: modules/tracking/src/trackerCSRTScaleEstimation.hpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ // and on any theory of liability, whether in contract, strict liability,
+ // or tort (including negligence or otherwise) arising in any way out of
+ // the use of this software, even if advised of the possibility of such damage.
+ //
+ //M*/
+
+#ifndef OPENCV_TRACKER_CSRT_SCALE_ESTIMATION
+#define OPENCV_TRACKER_CSRT_SCALE_ESTIMATION
+
+#include "precomp.hpp"
```
[the remark this answers is not in the document]
Comment (amuhic) [reply modules/tracking/src/trackerCSRTScaleEstimation.hpp]: I understand, though there seem to be trackers/files, already included in tracking module, that include precomp.hpp in their header file. (e.g. modules/tracking/src/tldDetector.hpp). Is there a reason for this?

**Comment by amuhic** [comment]
Thanks for the comments! 

I think I've fixed most of the problems you've pointed out. Except for the simple test part. Do you want me to add this tracker to 'test_trackers.cpp' or do you want me to send you a few or the tracking sequences, that I use for testing, together with a make file to build and test this?

**Comment by alalek** [comment]
Thank you!

I believe it is enough to add this tracker to 'test_trackers.cpp'.

Adding new datasets (tracking sequences) into OpenCV [repositories](https://github.com/opencv/opencv_extra/tree/3.4.0/testdata/cv/tracking) requires testdata approval (license), so it is better to reuse existing testdata (anyway you could add an external link on your dataset into Pull request description or documentation).

**Comment by alalek** [comment]
Looks better, Please take a look on current build issues on Windows/Android builders.
BTW, Identifiers started with underscore ('_') and capital letter are reserved (platform specific).

**Comment by moi90** [comment]
@amuhic I know this pull request got merged a long time ago, but would it be possible to follow this up with more documentation and comments? The tracker has a lot of parameters and (at least to me) it is not obvious what they do or how to tune them, even after reading the paper.

**Comment by moi90** [comment]
I created a draft pull request (#3300). Could someone help me improve it?

**Pull request merged**